### ISSUE TITLE
unit vectors

### ISSUE THREAD

**Issue by morningglory05** [jverzani/CalculusWithJulia.jl#51]
@jverzani 
Hi, thank you so much for creating this. I am currently using the arrow! function for 3D vectors. I did change the function since I needed to be able to specify a plot I wanted it on:
![image](https://user-images.githubusercontent.com/107124622/178520286-b7ace9af-149f-470f-82be-3f973f6a46d6.png)

I want to have vectors with uniform length. However, though normalizing the vectors changes the length (these vectors are pretty short and would be invisible without the normalization), they are not uniform.
![image](https://user-images.githubusercontent.com/107124622/178521582-12e722b7-6c1e-493e-bf0c-fcfeb4f0bf57.png)

My understanding of the arrow! code is that normalizing the vectors should work. However, I've only recently started coding. I'm not sure where I'm going wrong here. Thanks for any advice you give.

**Comment by jverzani** [comment]
If I had to guess, I'd hazard one that the scales of the x-y-z axes are not the same. In an x-y plot, we have the `aspect_ratio=:equal` setting, but if memory serves, that doesn't work in 3 dimensions. You could try limiting the plotting volume to see if things look more uniform then.

BTW, I should have that extra argument to `arrow!`. Let's leave this issue open.

**Comment by morningglory05** [comment]
You are completely right that the scales aren't equal. I can't believe I missed that! Thanks!

**Comment by jverzani** [comment]
I was off to fix this in the package, and find it has already been done:

```
function arrow!(plt, p, v; kwargs...)
  if length(p) == 2
      Plots.quiver!(plt, unzip([p])..., quiver=Tuple(unzip([v])); kwargs...)
  elseif length(p) == 3
      # 3d quiver needs support
      # https://github.com/JuliaPlots/Plots.jl/issues/319#issue-159652535
      # headless arrow instead
      #Plots.plot!(plt, unzip(p, p+v)...; kwargs...)
      ## use the above instead
      arrow3d!(plt, unzip(p,p+v)...; kwargs...)
  end
end
arrow!(p,v; kwargs...) = arrow!(Plots.current(), p, v; kwargs...)

```

Did you get code you copied from the notes? (If so, I got to fix those up...)

**Comment by morningglory05** [comment]
Yeah, I got it from the notes. I'm glad its all fixed up now. Thanks again for the quick and helpful responses.
